posts/basic-probabliity-in-lean.md

```diff
@@ -0,0 +1,241 @@
+
+If we define a measure `P` on `Ω`, we can talk about the law or distribution of `X`. It is by definition the map of the measure `P` by `X`, `P.map X`. There is no specific notation for that law.
+To say that `X` is Gaussian with mean 0 and variance 1, write `P.map X = gaussianReal 0 1`.
+
+The expectation of `X` is the integral of that function against the measure `P`, written `∫ ω, X ω ∂P`.
+The notation `P[X]` is shorthand for that expectation. In a `MeasureSpace`, we can further use the notation `𝔼[X]`.
+
+
+## Discrete probability
+
+In discrete probability, measurability is not an issue: every set and every function are measurable.
+The typeclass `[DiscreteMeasurableSpace Ω]` signals that every set of `Ω` is measurable and the lemma `MeasurableSet.of_discrete` provides a proof of measurability.
+To obtain measurability of a function from `Ω`, use `Measurable.of_discrete`.
+
+Any countable type with measurable singletons is a `DiscreteMeasurableSpace`. For example `ℕ` or `Fin n`.
+
+A way to define a probability measure on a discrete space `Ω` is to use the type `PMF Ω`, which stands for probability mass function.
+`PMF Ω` is the subtype of functions `Ω → ℝ≥0∞` that sum to 1.
+One can get a `Measure Ω` from `p : PMF Ω` with `p.toMeasure`.
+
+
+## Additional typeclasses on measurable spaces
+
+Some results in probability theory require the sigma-algebra to be the Borel sigma-algebra, generated by the open sets. For example, with the Borel sigma-algebra the open sets are measurable and continuous functions are measurable.
+For that we first need `Ω` to be a topological space and we then need to add a `[BorelSpace Ω]` variable.
+```lean
+variable {Ω : Type*} [MeasurableSpace Ω] [TopologicalSpace Ω] [BorelSpace Ω]
+```
+
+For properties related to conditional distributions, it is often convenient or necessary to work in a standard Borel space (a measurable space arising as the Borel sets of some Polish topology). See the `StandardBorelSpace` typeclass.
+Note that a countable discrete measurable space is a standard Borel space, so there is no need to worry about that typeclass when doing discrete probability.
+
+
+## Various probability definitions
+
+This section contains pointers to the Mathlib definitions for several probability objects.
+That list might be out of date when you read this! Look around in the [documentation](https://leanprover-community.github.io/mathlib4_docs/).
+
+### CDF, pdf, variance, moments
+
+Here is a list of common concepts about real probability distributions.
+
+- Probability density function of a random variable `X : Ω → E` in a space with measure `P : Measure Ω`, with respect to measure `Q : Measure E`: `pdf X P Q`.
+- Cumulative distribution function of `P : Measure ℝ`: `cdf P`. This is a `StieltjesFunction`, a monotone right continuous real function.
+- Expectation of `X` under `P`: `P[X]`.
```

Comment:
Could you expand on this a bit? I was trying some probability theory earlier this week, for the first time in at least a year, and the expectation notation was the only part I couldn't figure out. More specifically, have `P : Measure R` with `[MeasurableSpace R]`, and `[IsProbabilityMeasure P]`. I also have a function `R -> ENNReal`, which is measurable, and I'd like its average. I ended up using `lintegral` for this, but I expected (no pun intended) one of the expectation notations to work... 

So, could you clarify the types of `X` and `P` here, and which assumptions are needed for the notation to work?

Reply:
Thanks for the feedback. All the current expectation notations correspond to Bochner integrals and not Lebesgue, so they won't work for ENNReal.

I talk about the expectation in a bit more detail earlier in the post (in the random variable section): I'll add a remark over there about the two types of integrals and the assumptions on the spaces.

Reply:
I added a remark.

Reply:
Ah perfect, thanks!